scene/gui/tabs.cpp

```diff
@@ -593,6 +593,58 @@ void Tabs::remove_tab(int p_idx) {
 	_ensure_no_over_offset();
 }
 
+Variant Tabs::get_drag_data(const Point2 &p_point) {
+
+	int hover_now = -1;
+	for (int i = 0; i < tabs.size(); i++) {
+
+		if (i < offset)
+			continue;
+
+		Rect2 rect = get_tab_rect(i);
+		if (rect.has_point(p_point)) {
+			hover_now = i;
+		}
+	}
+
+	return hover_now;
+}
+
+bool Tabs::can_drop_data(const Point2 &p_point, const Variant &p_data) const {
+
+	int hover_now = -1;
+	for (int i = 0; i < tabs.size(); i++) {
+
+		if (i < offset)
+			continue;
+
+		Rect2 rect = get_tab_rect(i);
+		if (rect.has_point(p_point)) {
+			hover_now = i;
+		}
+	}
+
+	return (hover_now > -1);
+}
+
+void Tabs::drop_data(const Point2 &p_point, const Variant &p_data) {
+
+	int hover_now = -1;
+	for (int i = 0; i < tabs.size(); i++) {
```

Comment:
I'm looking at those three identical for loops. Normally I would create new function for that kind of things, but normally I'm not working on a files with many thousands of loc and  that amount of functions. It's good idea or bad?